src/Adapter/MSTest.TestAdapter/Discovery/AssemblyEnumerator.cs

```diff
@@ -411,6 +411,16 @@ private static bool ProcessITestDataSourceTests(UnitTestElement test, Reflection
     {
         foreach (ITestDataSource dataSource in testDataSources)
         {
+            // When reaching this point the global setup is to expand the parameterized tests but each data source
+            // could decide not to expand so we need to respect it.
+            if (dataSource is IExpandableDataSource { ExpandDataSource: false })
```

Comment:
Does it go the other way around? On assembly level it's set to not expand, but specific data sources can expand?

Reply:
Actually for that to work both ways, it needs to be nullable bool. Added another comment on that.

Reply:
You are right, I haven't considered the other way around. I'll think whether to rename to specify this is only to disable or expand behavior (there is probably some performance impact).